### PR TITLE
Update channels for updater server

### PR DESCRIPTION
**Pull request by MorrisJobke**
Followup to #15495 and #15498 

@skjnldsv This is what https://github.com/nextcloud/nextcloud-vue/pull/387 is needed for.

### CODE REVIEW

**Comment by MorrisJobke** [comment]
> @skjnldsv This is what [nextcloud/nextcloud-vue#387](https://github.com/nextcloud/nextcloud-vue/pull/387) is needed for.

Keep in mind that this should be back ported to 16 🙈

File: apps/updatenotification/src/components/root.vue [hunk 1]
```diff
@@ -236,6 +237,15 @@
 			channelList: function() {
 				let channelList = [];
 
+				channelList.push({
+					text: t('updatenotificaiton', 'Enterprise'),
+					longtext: t('updatenotification', 'For enterprise use. Provides always the latest patch level, but will not update to the next major release immediately. That update happens once Nextcloud GmbH has done additional hardening and testing for large-scale and mission-critical deployments. This channel is only available to customers and provides the Nextcloud Enterprise package. '),
```
Comment (nickvergessen) [inline apps/updatenotification/src/components/root.vue]:
```suggestion
					longtext: t('updatenotification', 'For enterprise use. Provides always the latest patch level, but will not update to the next major release immediately. That update happens once Nextcloud GmbH has done additional hardening and testing for large-scale and mission-critical deployments. This channel is only available to customers and provides the Nextcloud Enterprise package.'),
```

**Comment by MorrisJobke** [comment]
https://github.com/nextcloud/nextcloud-vue/pull/393 would also be nice.

**Comment by MorrisJobke** [comment]
But let's wait for #15506 to fix the regressions.

**Comment by MorrisJobke** [comment]
This is ready for review. What it does:

* it introduces a new channel called "enterprise"
* this channel is only available to be selected when a valid subscription is in place via the support app
* it adds a repair job that selects the enterprise channel if previously the production channel was selected and a valid subscription is available - otherwise it will go to the stable channel

**Review by nickvergessen — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**